CMakeLists.txt

```diff
@@ -3,108 +3,76 @@ project(drake-franka-driver)
-
-## end of setting environment variables
-
-
-configure_file(tests/data/franka_test.urdf "${CMAKE_CURRENT_BINARY_DIR}" COPYONLY)
-configure_file(tests/data/franka_test.yaml "${CMAKE_CURRENT_BINARY_DIR}" COPYONLY)
-
-# Find LCM
-list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake/modules")
-
-list(APPEND CMAKE_PREFIX_PATH ${DRAKE_BUILD_DIR})
-list(APPEND CMAKE_PREFIX_PATH ${DRAKE_BUILD_DIR}/lib/cmake)
-list(APPEND CMAKE_PREFIX_PATH ${DRAKE_BUILD_DIR}/lib/cmake/drake)
-
 list(APPEND CMAKE_PREFIX_PATH ${FRANKA_BUILD_DIR})
-
-find_package(drake REQUIRED)
 find_package(Franka REQUIRED)
-find_package(yaml-cpp REQUIRED COMPONENTS shared)
-
 message(STATUS "Franka_INCLUDE_DIRS: ${Franka_INCLUDE_DIRS}")
 message(STATUS "Franka_LIBRARIES: ${Franka_LIBRARIES}")
 
-find_package(lcm REQUIRED)
-find_package(gflags MODULE REQUIRED COMPONENTS shared)
+find_package(yaml-cpp REQUIRED COMPONENTS shared)
 find_package(GTest REQUIRED)
+find_package(gflags REQUIRED COMPONENTS shared)
+find_package(lcm REQUIRED)
 find_package(fcl REQUIRED)
 
-find_package( Python3 3.6 REQUIRED Interpreter Development NumPy)
+find_package(Python3 REQUIRED Interpreter Development NumPy)
 message(STATUS "Python3_VERSION: ${Python3_VERSION}")
 message(STATUS "Python3_INCLUDE_DIRS: ${Python3_INCLUDE_DIRS}")
 message(STATUS "Python3_LIBRARY_DIRS: ${Python3_LIBRARY_DIRS}")
 message(STATUS "Python3_LIBRARIES: ${Python3_LIBRARIES}")
 message(STATUS "Python3_NumPy_FOUND: ${Python3_NumPy_FOUND}")
 message(STATUS "Python3_NumPy_INCLUDE_DIRS: ${Python3_NumPy_INCLUDE_DIRS}")
 
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}")
-
-# message("Setting SPDLOG_INCLUDE_DIR; environment:" $ENV{SPDLOG_INCLUDE_DIR})
 if (DEFINED ENV{SPDLOG_INCLUDE_DIR})
     set(SPDLOG_INCLUDE_DIR $ENV{SPDLOG_INCLUDE_DIR})
     message(STATUS "SPDLOG_INCLUDE_DIR variable taken from environment: " ${SPDLOG_INCLUDE_DIR})
 else()
     set(SPDLOG_INCLUDE_DIR /opt/drake/include/spdlog)
     message(STATUS "SPDLOG_INCLUDE_DIR variable defaulted in CMakeList: " ${SPDLOG_INCLUDE_DIR})
 endif()
-# message(STATUS "SPDLOG_INCLUDE_DIR variable used by cmake: " ${SPDLOG_INCLUDE_DIR})
 
+# copy test data to build directory
+configure_file(tests/data/franka_test.urdf "${CMAKE_CURRENT_BINARY_DIR}" COPYONLY)
+configure_file(tests/data/franka_test.yaml "${CMAKE_CURRENT_BINARY_DIR}" COPYONLY)
 
-set(INCLUDES 
-    include
+# add include dir for Franka headers
+include_directories(
+    ${CMAKE_SOURCE_DIR}
     ${SPDLOG_INCLUDE_DIR}
     ${YAML_INCLUDES}
     ${FILESYSTEM_INCLUDE_DIR}
     ${Franka_INCLUDE_DIRS}
     ${Python3_INCLUDE_DIRS}
     ${Python3_NumPy_INCLUDE_DIRS}
     ${GTEST_INCLUDE_DIRS}
-) 
+)
 
-# add include dir for Franka headers
-include_directories(
-    ${INCLUDES}
+add_executable(send_stop
+    src/send_stop.cc
 )
 
+target_link_libraries(send_stop
+    lcm
+)
 
-set(SOURCES 
+# build the lcm driver that uses Franka Robot Interface (FRI)
+set(driver_name franka_driver)
```

Comment:
Is this used for anything?

Reply:
yes `franka_driver` is the main executable that runs when `franka.sh` is called